src/cargo/core/resolver/mod.rs

```diff
@@ -687,7 +721,7 @@ impl Context {
                                              feature)));
                 }
             }
-            ret.push((dep, base));
+            ret.push((Rc::new(dep.clone()), base));
```

Comment:
Because of the `set_foo` methods on `Dependency`, having it store an `Arc<DependencyInner>` doesn't seem to work that well. Every use of `set_foo` would have to copy the whole `Dependency`. I'm most of the way through just switching all the non-creating uses over to `Rc<Dependency>`, but because of `Summary::map_dependencies` I'm using the experimental `Rc::try_unwrap` to avoid unnecessary copies. Let me know if you see a better way to do this.